src/core/Form/FilterInput/FilterInput.tsx

```diff
@@ -0,0 +1,186 @@
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  inputElement: `${baseClassName}_input`,
+};
+
+type FilterInputStatus = Exclude<InputStatus, 'success'>;
+
+export interface FilterInputProps<T>
+  extends Omit<HtmlInputProps, 'type'>,
+    TokensProp {
+  /** FilterInput container div class name for custom styling. */
+  className?: string;
+  /** FilterInput container div props */
+  inputContainerProps?: Omit<HtmlDivProps, 'className'>;
+  /** Disable input usage */
+  disabled?: boolean;
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: LabelMode;
+  /** Text to mark a field optional. Will be wrapped in parentheses and shown after labelText. */
+  optionalText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: FilterInputStatus;
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** FilterInput name */
+  name?: string;
+  /** Align label on top or on the left side of the input field
+   * @default 'top'
+   */
+  labelAlign?: 'top' | 'left';
+  /** Items to be filtered */
+  items: Array<T>;
+  /** Returns the filtered items */
+  onFiltering: (filteredItems: Array<T>) => void;
+  /** Filtering rule to be used */
+  filterRule: (item: T, query: string) => boolean;
+}
+
+class BaseFilterInput<T> extends Component<FilterInputProps<T>> {
+  render() {
+    const {
+      className,
+      inputContainerProps,
+      visualPlaceholder,
+      labelText,
+      labelMode,
+      optionalText,
+      status,
+      statusText,
+      id,
+      labelAlign,
+      'aria-describedby': ariaDescribedBy,
+      items: propItems,
+      onFiltering: propOnFiltering,
+      filterRule: propFilterRule,
+      ...passProps
+    } = this.props;
+
+    const onChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
+      const { items, onFiltering, filterRule } = this.props;
+      const { value } = event.target;
+
+      const filteredItems: T[] = items.reduce((filtered: T[], item: T) => {
+        if (filterRule(item, value)) {
+          filtered.push(item);
+        }
+        return filtered;
+      }, []);
+
+      onFiltering(filteredItems);
+    };
+
+    const statusTextId = statusText ? `${id}-statusText` : undefined;
+
+    return (
+      <HtmlDiv
+        {...inputContainerProps}
+        className={classnames(baseClassName, className, {
+          [filterInputClassNames.disabled]: !!passProps.disabled,
+          [filterInputClassNames.error]: status === 'error',
+        })}
+      >
+        <HtmlDiv
+          className={classnames(filterInputClassNames.wrapper, {
+            [filterInputClassNames.labelAlignLeft]: labelAlign === 'left',
+          })}
+        >
```

Comment:
This is the additional wrapper what we have discussed. It tries to keep the outside container to not affect the component layout (e.g using `display`).

Reply:
Things that need to remain unchanged e.g. display, possibly max-width etc. should probably be specified for the wrapper rather than the input container?

Reply:
`display` and `width` is now moved to be told for the wrapper in baseStyles.